lib/streamlit/elements/json.py

```diff
@@ -106,18 +82,11 @@ def json(
             body = body.to_dict()
 
         if not isinstance(body, str):
-            # Check if body is iterable, if it is, convert its sets to lists
-            if is_iterable(body):
-                # body is iterable, look for sets and change them to lists
-                # TODO(harahu): This function does not handle iterables in
-                #  general. Either generalize the function, or do further type
-                #  checking here.
-                body = _convert_sets_to_lists(body)  # type: ignore[arg-type]
-
             try:
-                # Serialize body to string
-                body = json.dumps(body, default=repr)
-            except TypeError as err:
+                # Serialize body to string and try to interpret sets as lists
+                body = json.dumps(body, default=_convert_sets_to_lists_repr)
```

Comment:
I think we need to use the same default function also in the second `json.dumps` (line 94)